cunumeric/array.py

```diff
@@ -1741,6 +1741,82 @@ def astype(
         result._thunk.convert(self._thunk, warn=False)
         return result
 
+    def take(self, indices, axis=None, out=None, mode="raise"):
+        """a.take(indices, axis=None, out=None, mode="raise")
+
+        Take elements from an array along an axis.
+
+        Refer to :func:`cunumeric.take` for full documentation.
+
+        See Also
+        --------
+        cunumeric.take : equivalent function
+
+        Availability
+        --------
+        Multiple GPUs, Multiple CPUs
+
+        """
+        if not np.isscalar(indices):
+            # if indices is a tuple or list, bring sub-tuples to the same shape
+            # and concatenate them
+            indices = convert_to_cunumeric_ndarray(indices)
+
+        if axis is None:
+            self = self.ravel()
+            axis = 0
+        elif axis < 0:
+            axis = self.ndim + axis
+        if axis is not None and axis >= self.ndim:
```

Comment:
`if axis < 0 or axis >= self.ndim`

Reply:
right, I didn't think about the case when `axis<-self.ndim`